### ISSUE TITLE
[iOS 9] Add Universal Links support.

### ISSUE THREAD

**Issue by alloy** [artsy/eigen#767]
This makes it so that Eigen will open artsy URLs instead of in Mobile Safari.

This requires a JSON file to be served from the root of the `artsy.net` domain and over HTTPS, see https://developer.apple.com/library/prerelease/ios/documentation/General/Conceptual/AppSearch/UniversalLinks.html.

@joeyAghion @dblock Should I make a PR for Gravity that adds the JSON file to `public` so it can be statically served?

- [x] ~~Define list of paths we want to support.~~ Because of the `/:profile_id` route, everything will be watched anyways, so there’s no point in doing this. The app will just accept all and show a bunch of routes in a webview only.
- [x] Make naked `artsy.net` domain work.

**Comment by orta** [comment]
I think it would need to go on force

**Comment by alloy** [comment]
@orta Not technically, but I guess you mean because Force serves `www.`? And then add an extra file on Martsy for `m.artsy.net`? I guess it makes sense over serving it from `api`, it would just be nice to not have to maintain practically the same file in multiple locations, so maybe both from Force?

**Comment by orta** [comment]
they both could redirect to a file on gravity to reduce duplication, but I think a reference would be needed on both.

**Comment by alloy** [comment]
> but I think a reference would be needed on both

For what reason? It’s not required for the features to work, Apple only wants you to have separate files for different content being served, Force and Martsy both serve the same content.

**Comment by orta** [comment]
I thought it cared about the exact url prefixes, `m.` and `www.` but you've probably read it better.

**Comment by alloy** [comment]
I might be wrong, but that’s what I gathered from this:

> For example, apple.com and developer.apple.com need separate apple-app-site-association files, because these domains serve different content. In contrast, apple.com and www.apple.com don’t need separate site association files—because both domains serve the same content—but both domains must make the file available.

**Comment by orta** [comment]
> but both domains must make the file available

I think, in this case because it's not just a http redirect like the `www.` would be it would require having it in both places. Conceptually it is "different content" as it's an entirely different rendering path.

**Comment by alloy** [comment]
Need to limit the URL paths that we support with Universal Links to those for which we actually have views. I’m thinking only these from the switchboard:

* `/artist/:id`
* `/:profile_id/artist/:id`
* `/artwork/:id`
* `/gene/:id`
* `/show/:id`
* `/:profile_id/for-you`
* `/:profile_id/browse/artist/:id`
* `/favorites`
* `/browse`
* `/categories`
* `/:profile_id`
* `/works-for-you`

@orta Any others that I missed or ones that we don’t want to catch at all?

**Comment by alloy** [comment]
Oh, we’re missing `/shows` and `/articles` in the switchboard, should add those there and to this list.

**Comment by orta** [comment]
I think that's pretty legit - the :profile ones are pretty much catch-alls that should work well for us

**Comment by alloy** [comment]
Oh, actually the `/:profile_id` should be removed, as it would catch anything.

**Comment by alloy** [comment]
```
/browse
/favorites
/works-for-you
/articles
/artist/:id
/artwork/:id
/categories
/gene/:id
/shows
/show/:id
/:profile_id/for-you
/:profile_id/artist/:id
/:profile_id/browse/artist/:id
```

**Comment by orta** [comment]
We really do need that `:profile_id` I'm afraid - can't link to fairs/auctions/users etc - https://www.artsy.net/fnb-joburgartfair-2015

**Comment by orta** [comment]
we can also handle `/feature/:feature_id` - https://www.artsy.net/feature/west-bund-art-and-design-2015 - transparently via martsy/force

**Comment by orta** [comment]
and we definitely want `/auction/:auction_id` - https://www.artsy.net/auction/she-inspires-art

Might be worth doing a `rake routes` in force/martsy

**Comment by alloy** [comment]
> we can also handle `/feature/:feature_id` - https://www.artsy.net/feature/west-bund-art-and-design-2015 - transparently via martsy/force

You mean that we’d show it in an internal webview, right?

**Comment by orta** [comment]
Yep

**Comment by alloy** [comment]
Fixed by #786.
